internal/usecase/repository/host_repository.go

```diff
@@ -31,12 +33,12 @@ func (r *hostRepository) MatchDomain(db *gorm.DB, options *interactor.HostConfOp
 		return nil, internal_errors.NilArgError("options")
 	}
 
-	// look through domains and find domains with non-NULL token
 	// TODO: match FQDN with domain realms
 	var domains []model.Domain
 	tx := db.Model(&model.Domain{}).
 		Joins("left join ipas on domains.id = ipas.id").
-		Where("domains.org_id = ?", options.OrgId)
+		Where("domains.org_id = ?", options.OrgId).
+		Where("domains.auto_enrollment_enabled = TRUE")
```

Comment:
This change will result in the same error message when a domain is not found and a domain is disabled. For debugging and logging, it would be better to have different error results for "no domain found" and "domain is disabled".

Reply:
OK, I'll change the approach, but it will involve no longer limiting the number of records in the response set.  I think that's OK, we expect most orgs to have a small number, and we don't expect any to have a huge number of registered services.

Reply:
Besides, when it comes to "semantic" matching of the host domain name to choose the best candidate domain, we will have to iterate a set of records.  Because SQL is probably not expressive enough for what we would do.

Reply:
We have not designed rules to match a FQDN to a domain. We'll probably need another table and additional APIs to create matching rules.

Reply:
I think we can hardcode matching rules with whatever heuristic(s) we think make sense.  And when customers beg us for something more flexible or different, *then* we work out whether we should do it, and if so, how :)